Add test to validate the cpu affinity change with load

sathnaga opened the pull request:
This patch adds a test to validate the host cpu statistics
with respect to affinity changes while guest under load.

Signed-off-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>

sathnaga commented:
Dependancy PR: https://github.com/avocado-framework/avocado-vt/pull/1243

sathnaga commented:
```
 (1/3) guest_import.qemu.qcow2.virtio_scsi.smp2.virtio_net.HostOS.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native: PASS (49.67 s)
 (2/3) guest_cpu.vcpupin.qemu.qcow2.virtio_scsi.smp2.virtio_net.HostOS.ppc64le.powerkvm-libvirt.guestpin.without_emulatorpin.randompin.positive.with_load_switch: PASS (155.24 s)
 (3/3) guest_remove.qemu.qcow2.virtio_scsi.smp2.virtio_net.HostOS.ppc64le.powerkvm-libvirt.remove_guest.without_disk: PASS (3.30 s)
```

chunfuwen commented on libvirt/tests/cfg/cpu/guestpin.cfg:
```diff
@@ -43,13 +43,25 @@
                     limit_vcpu_threads = 1
                     limit_vcpu_sockets = 1
                     condn = "hotplug"
+                - with_load_switch:
```
maybe filter case here by adding only , not indiviually add no key word in variant sequential and emulatorpin

sathnaga replied:
Sure

chunfuwen commented on libvirt/tests/src/cpu/guestpin.py:
```diff
@@ -120,26 +120,65 @@ def set_condition(vm_name, condn, reset=False, guestbt=None):
+
+            logging.debug("Check cpustats for emulatorpinned cpu")
+            if cpustats[cpus_list[-1]][0] > 0:
+                fail = True
```
I understand that you want to use one boolean fail to accumulate result(we may call it soft fail in other way),but it bring additional efforts to do troubleshooting. So suggest still use test.fail() if failed at any certain point.

sathnaga replied:
@chunfuwen There is proper logging errors in each soft fail place.I felt that will suffice the tester to look back/troubleshoot?, and these checks are not any event failure so test case flow should not affect.

sathnaga commented:
@chunfuwen Addressed review comments,
just did a dry-run to check the config changes..
```
(1/3) guest_import.qemu.qcow2.virtio_scsi.smp2.virtio_net.HostOS.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native: CANCEL (0.01 s)
 (2/3) guest_cpu.vcpupin.qemu.qcow2.virtio_scsi.smp2.virtio_net.HostOS.ppc64le.powerkvm-libvirt.guestpin.with_emualorpin.randompin.positive.with_load_switch: CANCEL (0.01 s)
 (3/3) guest_remove.qemu.qcow2.virtio_scsi.smp2.virtio_net.HostOS.ppc64le.powerkvm-libvirt.remove_guest.without_disk: CANCEL (0.01 s)
```
